### DISCUSSION TITLE
How to upgrade argo-cd using manual install?

### DISCUSSION THREAD

**Discussion by gajus** [argoproj/argo-cd#5310] [Q&A]
Currently, any time that I upgrade argo-cd I need to perform:

```bash
kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj/argo-cd/stable/manifests/install.yaml
kubectl apply -f config-map.yaml
kubectl apply -f ingress.yaml
# Not clear why `repocreds` are deleted every time `argo-cd` is upgraded.
argocd repocreds add https://gitlab.com/contrawork --username <username> --password <password>

```

Is there a more streamlined method?

e.g. Is there a helm chart for argo-cd?

**Comment by jannfis** [comment]
I would suggest to use Kustomize with a remote base, and providing all of your specific configuration (such as, configmaps and secrets) as overlays (or strategic merge patches). In a simple case, it is not even very complicated to use.

So, in a dedicated directory, copy your YAML files to it and create a `kustomization.yaml` with a content similar to:

```yaml
resources:
- https://github.com/argoproj/argo-cd/manifests/cluster-install?ref=stable
- config-map.yaml
- ingress.yaml
```

And then run `kustomize build . | kubectl -n argocd apply -f -` (or redirect output of `kustomize build .` to a file, to check the generated YAML before applying to your cluster).

You can also replace `?ref=stable` in the example with the version you want to upgrade to, i.e. `?ref=v1.8.3` 

There are more cool things you can do with Kustomize, and it doesn't have a very steep learning curve.

**Reply by gajus** [reply]
The bigger problem is the:

```
argocd repocreds add https://gitlab.com/contrawork --username <username> --password <password>
```

It is not clear why repocreds are deleted every time I apply a patch.

**Reply by jannfis** [reply]
Are the repocreds defined in the ConfigMap you apply according to your workflow?
